modules/sdk-coin-icp/src/icp.ts

```diff
@@ -141,8 +143,81 @@ export class Icp extends BaseCoin {
     return true;
   }
 
-  async isWalletAddress(params: TssVerifyAddressOptions): Promise<boolean> {
-    return this.isValidAddress(params.address);
+  /**
+   * Verify that an address belongs to this wallet.
+   *
+   * @param {TssVerifyIcpAddressOptions} params - Verification parameters
+   * @returns {Promise<boolean>} True if address belongs to wallet
+   * @throws {InvalidAddressError} If address format is invalid
+   * @throws {Error} If invalid wallet version or missing parameters
+   */
+  async isWalletAddress(params: TssVerifyIcpAddressOptions): Promise<boolean> {
+    const { address, rootAddress, walletVersion } = params;
+
+    if (!this.isValidAddress(address)) {
+      throw new InvalidAddressError(`invalid address: ${address}`);
+    }
+
+    if (walletVersion === 1) {
+      return this.verifyMemoBasedAddress(address, rootAddress);
```

Comment:
even in case of memo based address, root address still needs to be verified against they common keychain